js/queue/queue-data-structure.js

```diff
@@ -9,18 +9,47 @@ class QueueDataStructure {
   }
 
   canEnqueue() {
-    // ... your code goes here
+
+    if (this.queueControl.length < this.MAX_SIZE) return true
+    else return false
+
   }
 
   isEmpty() {
-    // ... your code goes here
+
+    if (this.queueControl.length === 0) return true
+    else return false
+
   }
 
   enqueue(item) {
-    // ... your code goes here
+
+    if (this.canEnqueue()) {
+
+      this.queueControl.push(item)
```

Comment:
```suggestion
        this.queueControl.unshift(item);
```
Recuerda que en la estructura FIFO el primer elemento insertado, será el primer elemento eliminado, por lo que un método shift() seria util, el elimina el primer elemento del array y lo retorna.